Allow the validator to check for an URL

neveldo opened the pull request:
For now, the validator tool is only able to check for HTML files through the file system. This PR  aims to provide a way to check also URLs, so users can run this kind of commands : `dist/validate https://www.ampproject.org/how-it-works/` .

cramforce commented:
Thanks!!! Assigning to @powdercloud

neveldo commented:
You are welcome !

PS : I just signed the Contributor License Agreement .

powdercloud commented on validator/build.py:
```diff
@@ -194,30 +194,58 @@ def GenerateValidateBin(out_dir):
-        if (output[0] === 'PASS') {
-          for (var i = 0; i < output.length; ++i) {
-            console.info(output[i]);
+        var validateFile = function(contents, filename) {
```
Can you make this its own function please? As in, just like main is a function, have function validateFile? I think it'll be easier to read that way, less nesting.

powdercloud commented on validator/build.py:
```diff
@@ -194,30 +194,58 @@ def GenerateValidateBin(out_dir):
+        };
+
+        if (full_path.indexOf('http://') === 0 || full_path.indexOf('https://') === 0) {
+            var callback = function(response) {
```
Please try doing indent by two spaces, for consistency. This may be a Google thing, but as is the code looks a little ragged, e.g., the '}' in 244 and 248 don't line up either. Sorry for the nitpicking!

powdercloud commented on validator/build.py:
```diff
@@ -194,30 +194,58 @@ def GenerateValidateBin(out_dir):
+              var contents = '';
+
+              response.on('data', function (chunk) {
+                contents += chunk;
```
I honestly don't know how much a difference it makes in practice, but think that the usual approach would be to accumulate the chunks in some array and join at the end to make sure we don't get quadratic concatenation performance on large files (and shitty Javascript interpreters).
var chunks = [];
...
    chunks.push(chunk);
...
validateFile(chunks.join(''), ...

powdercloud commented:
@neveldo Thanks a lot, I nitpicked a couple things the rest looks cool to me and certainly useful. Happy to take another look when it's done.

neveldo commented:
@powdercloud Thanks for your code review ! I hope this little improvement will be useful for the editors in order to validate their online AMP pages too.

My apologies for the indentation inconsistency, I have fixed it. I also moved the validateFile() function out of the main() function and the chunks are now accumulated into an array, instead of being concatened into a string.

powdercloud commented:
@neveldo Thanks a lot!
